### PR TITLE
docs: Align authentication provider approach for all versions

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Relates to https://github.com/kedacore/keda-docs/issues/711

### CODE REVIEW

File: content/docs/2.5/concepts/authentication.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ metadata:
   namespace: default # must be same namespace as the ScaledObject
 spec:
   podIdentity:
-      provider: none | azure | gcp | aws-eks | aws-kiam  # Optional. Default: none
```
Comment (tomkerkhove) [inline content/docs/2.5/concepts/authentication.md:95]: This is how I noticed it but it was not in 2.6 -> 2.9 docs.

**Comment by tomkerkhove** [comment]
Any thoughts @kedacore/keda-maintainers? (I'll fix merge conflicts)

**Comment by JorTurFer** [comment]
It looks awesome, but the dropdown doesn't work, it points to wrong url (missing `/authentication-providers/`)
image

**Comment by tomkerkhove** [comment]
I think that's out of scope for this PR as I'm just adding data and not changing the workings for that, will open an issue for it - OK @JorTurFer ?

**Comment by tomkerkhove** [comment]
Redeployed and it's fine: https://app.netlify.com/sites/keda/deploys/632961ba6735dc53e7b3d68a

**Comment by JorTurFer** [comment]
> I think that's out of scope for this PR as I'm just adding data and not changing the workings for that, will open an issue for it - OK @JorTurFer ?

It's okey to me

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**